redux/redux.d.ts

```diff
@@ -1,50 +1,79 @@
-// Type definitions for Redux v1.0.0
+// Type definitions for Redux v3.0.0
 // Project: https://github.com/rackt/redux
-// Definitions by: William Buchwalter <https://github.com/wbuchwalter/>, Vincent Prouillet <https://github.com/Keats/>
+// Definitions by: William Buchwalter <https://github.com/wbuchwalter/>,
+//                 Vincent Prouillet <https://github.com/Keats/>,
+//                 Michael Tiller <https://github.com/xogeny/>
 // Definitions: https://github.com/borisyankov/DefinitelyTyped
 
 declare module Redux {
+    // Minimalist Action definition
+    interface Action {
+        type: string;
+    }
 
+    // Not used...?
     interface ActionCreator extends Function {
-        (...args: any[]): any;
+        (...args: any[]): Action;
+    }
+
+    // Also not used here...
+    interface StoreMethods<S,A extends Action> {
+        dispatch: Dispatch;
+        getState(): S;
     }
 
-    interface Reducer extends Function {
-        (state: any, action: any): any;
+    // Reducer is a function that takes a state of type S and an
+    // action and returns a new state, also of type S
+    interface Reducer<S> extends Function {
+        (state: S, action: any): S;
     }
 
+    // Dispatch is a function that takes an action of type A and
+    // returns an instance of that same action (still of type A)
     interface Dispatch extends Function {
-        (action: any): any;
+        <A extends Action>(action: A): any;
     }
 
-    interface StoreMethods {
+    // MiddlewareArg is an instance of middleware that provides a
+    // dispatch method and a getState method.  The type parameter S is
+    // the type of the state.  No type parameter is provided on
+    // Dispatch because the middleware generally needs to be able to
+    // handle any type of action.
+    interface MiddlewareArg<S> {
         dispatch: Dispatch;
-        getState(): any;
+        getState: () => S;
     }
 
-
-    interface MiddlewareArg {
-        dispatch: Dispatch;
-        getState: Function;
+    // Middleware is constructed from an instance of MiddlewareArg<S> where
+    // S is the type of the state.
+    interface Middleware<S> extends Function {
+        (obj: MiddlewareArg<S>): Function;
     }
 
-    interface Middleware extends Function {
-        (obj: MiddlewareArg): Function;
+    // Store manages the state of the system.  The state of the system
+    // should be of type S.
+    class Store<S> {
+        getReducer(): Reducer<S>;
```

Comment:
@xogeny this is ok but the other comments are imo not necessary and can be fixed by other PRs if needed, this PR should be merged asap